src/api/app/models/workflow.rb

```diff
@@ -47,27 +50,27 @@ def steps
   end
 
   def filters
-    filters = {}
-
-    return filters if supported_filters.blank?
+    return {} if supported_filters.blank?
 
-    SUPPORTED_FILTERS.each do |filter|
-      SUPPORTED_FILTER_TYPES.each do |filter_type|
-        # The filter type might not be present... so in that case, we go to the next
-        next unless (filter_values = supported_filters.dig(filter, filter_type))
+    @filters ||= SUPPORTED_FILTERS.each_with_object({}) do |filter, filters_hash|
+      if filter == :event
+        filters_hash[filter] = supported_filters[filter]
+      else
+        SUPPORTED_FILTER_VALUES.each do |filter_value|
```

Comment:
I think the same (precedence of `only` over `ignore`) we do already in the steps. 

https://github.com/openSUSE/open-build-service/blob/master/src/api/app/models/workflow/step.rb#L108-L110

Reply:
From line 12: `SUPPORTED_FILTER_VALUES = [:only, :ignore].freeze` should be enough.

I'm missing something. What are you suggesting?

Reply:
This loop and the link that @hennevogel shared implement the same logic. We don't need this logic duplication. Let's remove it in the `Workflow` model.